openhands_aci/editor/editor.py

```diff
@@ -275,6 +268,70 @@ def insert(
         success_message += 'Review the changes and make sure they are as expected (correct indentation, no duplicate lines, etc). Edit the file again if necessary.'
         return CLIResult(output=success_message)
 
+    def delete(self, path: Path, lines_range: list[int]) -> CLIResult:
+        """
+        Deletes text content in file from the given range.
+        """
+        try:
+            file_text = self.read_file(path)
+        except Exception as e:
+            raise ToolError(f'Ran into {e} while trying to read {path}') from None
+
+        file_text = file_text.expandtabs()
+
+        file_text_lines = file_text.split('\n')
+        num_lines = len(file_text_lines)
+
+        self._validate_range(lines_range, num_lines)
+
+        start_line, end_line = lines_range  # inclusive
+
+        new_file_text_lines = (
+            file_text_lines[:start_line] + file_text_lines[end_line + 1:]
+        )
+        snippet_lines = (
+            file_text_lines[max(0, start_line - SNIPPET_CONTEXT_WINDOW) : start_line]
+            + file_text_lines[
+                end_line + 1 : min(num_lines, end_line + SNIPPET_CONTEXT_WINDOW)
+            ]
+        )
+        new_file_text = '\n'.join(new_file_text_lines)
+        snippet = '\n'.join(snippet_lines)
+
+        self.write_file(path, new_file_text)
+        self._file_history[path].append(file_text)
+
+        success_message = f'The file {path} has been edited. '
+        success_message += self._make_output(
+            snippet,
+            'a snippet of the edited file',
+            # max(1, start_line - SNIPPET_CONTEXT_WINDOW + 1),
+        )
+
+        success_message += 'Review the changes and make sure they are as expected (correct indentation, no duplicate lines, etc). Edit the file again if necessary.'
+        return CLIResult(output=success_message)
+
+    def move_code_block(
+            self, from_file: Path, from_range: list[int], dst_file: Path, insert_line: int
+        ) -> CLIResult:
+        """
+        Move a block of code from one file to another.
+        """
+        file_content = self.read_file(from_file)
+        file_content_lines = file_content.split('\n')
+        start_line, end_line = from_range
+        code_block = '\n'.join(
+            file_content_lines[start_line:]
+            if end_line == -1
+            else file_content_lines[start_line: end_line + 1]
+        )
+        delete_result = self.delete(from_file, from_range)
+        insert_result = self.insert(dst_file, insert_line, code_block, True)
+
+        return CLIResult(
+            output=f'Code block moved from {from_file} to {dst_file}.\n{delete_result.output}\n{insert_result.output}'
+        )
+
     def validate_path(self, command: Command, path: Path) -> None:
```

Comment:
I think we may need to do validation for the command to make sure it's not a directory?

Reply:
Yes, we are doing that above.